### PR TITLE
chakrashim: relocate inspector code

### PR DESCRIPTION
**Pull request by kfarnung**
* Removed the "inspector" subdirectory from chakrashim
* Integrated the build into chakrashim.gyp
* Cleaned up the node.gyp/node.gypi files
* Enabled native inspector tests

##### Checklist

- [x] `make -j4 test` (UNIX), or `vcbuild test` (Windows) passes
- [x] commit message follows [commit guidelines](https://github.com/nodejs/node/blob/master/CONTRIBUTING.md#commit-guidelines)

##### Affected core subsystem(s)

chakrashim

### CODE REVIEW

**Comment by agarwal-sandeep** [comment]
If CI is green this looks good to me

**Comment by kfarnung** [comment]
Added a few more changes to handle issues on Linux (still needs to be re-verified on Windows).  Will squash changes before commit.

CI:  https://ci.nodejs.org/job/chakracore-test-pull-request/29/

**Comment by kunalspathak** [comment]
>         '.',

Was this added for building inspector? Is this still needed?

---
Refers to: deps/chakrashim/chakrashim.gyp:23 in ab263b4. [](commit_id = ab263b49dbff9eeae4e9dc060a48c0073790ca57, deletion_comment = False)

File: node.gypi [hunk 1]
```diff
@@ -256,20 +239,13 @@
         }],
       ],
     }],
-    [ 'node_engine=="v8"', {
+    ['node_engine=="chakracore"', {
```
Comment (kunalspathak) [inline node.gypi:242]: >node_engine [](start = 6, length = 11)

can you try building v8 as well? There is some known float64 issue, but with your change, we should atleast not see more issues.

Comment (kfarnung) [reply node.gypi:242]: I've been building V8 and ChakraCore using both Linux and Windows.  They both build and the tests pass with some small amount of flakiness.

**Comment by kfarnung** [comment]
@kunalspathak that include path is currently still needed, but may be able to be eliminated in the future.

File: test/parallel/test-repl.js [hunk 2]
```diff
@@ -224,28 +213,28 @@ function error_test() {
       send: '(function() { "use strict"; return 0755; })()',
       expect: common.engineSpecificMessage({
         v8: /\bSyntaxError: Octal literals are not allowed in strict mode/,
-        chakracore: cc_re4})
+        chakracore: /^SyntaxError: Octal numeric literals and escape characters not allowed in strict mode/}) // eslint-disable-line max-len
```
Comment (kunalspathak) [inline test/parallel/test-repl.js:216]: curious, why is this changed?

Comment (kfarnung) [reply test/parallel/test-repl.js:216]: Pulling these regexes out was making them harder to maintain.  Since they have to ignore the length of the lines upstream I didn't see the point in maintaining these exceptions.  In one case the V8 test was failing purely because the regex was formed incorrectly.

File: Makefile [hunk 3]
```diff
@@ -195,7 +195,7 @@ test: all
 	$(MAKE) build-addons
 	$(MAKE) build-addons-napi
 	$(MAKE) cctest
-	$(PYTHON) tools/test.py --mode=release -J \
+	$(PYTHON) tools/test.py --mode=release --flaky-tests=$(FLAKY_TESTS) -J \
```
Comment (kunalspathak) [inline Makefile:198]: This change LGTM. 

//cc: @joaocgreis  in case he has any objection. To give you context, this was to not see flaky test result in our buddy testing on linux.

File: test/parallel/test-repl.js (hunk 2 above)
Comment (kunalspathak) [reply test/parallel/test-repl.js:216]: Got it. Makes sense.

**Review by kunalspathak — APPROVED** [review]
